care/users/api/viewsets/users.py

```diff
@@ -286,3 +286,17 @@ def pnconfig(self, request, *args, **kwargs):
                 setattr(user, field, request.data[field])
         user.save()
         return Response(status=status.HTTP_200_OK)
+
+class CheckUsernameAPIView(GenericAPIView):
+    """
+    Checks availability of username by getting as query, returns 200 if available, and 409 otherwise.
+    """
+
+    permission_classes = (AllowAny, )
```

Comment:
Can we switch this to authenticated users only? Having an open endpoint to guess usernames might be a potential attack vector. Since users creation is an authenticated process for us, it makes sense in that way as well.